simulation_parameters/SimulationParameters.cs

```diff
@@ -177,17 +178,16 @@ public bool DoesCompoundExist(string name)
     /// </summary>
     public List<Compound> GetCloudCompounds()
     {
-        var result = new List<Compound>();
-
-        foreach (var entry in compounds)
-        {
-            if (entry.Value.IsCloud)
-            {
-                result.Add(entry.Value);
-            }
-        }
+        return compounds.Values.Where(c => c.IsCloud).ToList();
+    }
 
-        return result;
+    /// <summary>
+    ///   Returns all environmental *molecular* compounds that are dissolved in the environment, i.e. gas.
+    /// </summary>
+    /// <remarks>This excludes sunlight, and includes O2, CO2, N...</remarks>
+    public List<Compound> GetGasCompounds()
+    {
+        return compounds.Values.Where(c => c.IsCloud && c.IsEnvironmental && c.CanBeDistributed).ToList();
```

Comment:
I feel a bit confused by these flag checks here. It *should* be the case that IsCloud and IsEnvironmental can't be true for the same compound. Also CanBeDistributed is related to colony compound sharing, so this confuses me a lot.

Reply:
I agree that by the JSON compounds, this should return an empty list, yet since changes are applied, it probably doesn't... I'm as confused as you on this, I don't recall why I used `IsCloud`... I'll have to test this when I'll have a moment.

`CanBeDistributed` is used to exclude sunlight. That was both the best heuristic I had at hand, and something logical, since globally "gas compounds" should be spread in colonies, unlike whatever compound type is sunlight.

Reply:
Perhaps if there isn't a good existing flag for it you should add a new property to the Compound class that specifies that it is a gas?

Reply:
I guess it would make sense to have a good flag for sunlight (and potential other similar compounds), but I'm short for ideas at the moment. Unless someone comes by with a great idea, I'll simply add a "isSunlight" flag whenever I have time to wrap everything up.

Reply:
Why not "IsGas"?

Reply:
For some  reason those flags are never simultaneously true?